sys/kern/cred.c

```diff
@@ -1,9 +1,21 @@
 #include <sys/cred.h>
 #include <sys/proc.h>
 #include <sys/libkern.h>
+#include <sys/errno.h>
 #include <sys/mutex.h>
 
 void cred_fork(proc_t *to, proc_t *from) {
   assert(mtx_owned(&from->p_lock));
   memcpy(&to->p_cred, &from->p_cred, sizeof(cred_t));
 }
+
+int cred_cansignal(proc_t *p, cred_t *cred) {
+  assert(mtx_owned(&p->p_lock));
+
+  if (cred->cr_ruid != p->p_cred.cr_ruid &&
+      cred->cr_ruid != p->p_cred.cr_suid &&
+      cred->cr_euid != p->p_cred.cr_ruid && cred->cr_euid != p->p_cred.cr_suid)
```

Comment:
I assume the semantics of this check are based on some standard or an existing implementation in some other kernel -- can you point me to it so that I can compare the two?

Reply:
Yes. It is described in standard: [kill()](https://pubs.opengroup.org/onlinepubs/009695399/). And [example](http://bxr.su/FreeBSD/sys/kern/kern_prot.c#1552) in code of FreeBSD.

Reply:
Thanks, looks good!